### PR TITLE
Doesn't select values for missing Month and Day in EDTF dates

### LINKED ISSUE #963: Publication and Creation dates with just years should not have months and days filled in automatically in the form
I entered only years for publication and creation dates as follows:

I saved this as a draft and the item view page displayed just these years as it should:

But when I opened up the form to continue editing my draft, the dates are filled in this way:

This should only have the years, but January 1 has been filled in as well in both cases.

### PR DESCRIPTION
**Pull request by jermnelson**
## Why was this change made?
Fixes #963 

## How was this change tested?
New component tests

## Which documentation and/or configurations were updated?
n/a

### CODE REVIEW

File: spec/components/works/dates_component_spec.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@
     end
   end
 
-  context 'with a populated form with a single craeation_date' do
+  context 'with a populated form with a single creation_date' do
```
Comment (mjgiarlo) [inline spec/components/works/dates_component_spec.rb:41]: :clap:

File: spec/components/works/dates_component_spec.rb [hunk 2]
```diff
@@ -50,4 +50,26 @@
...
     end
   end
+
+  context 'with a populated form containing only year for single creation_date' do
+    let(:year_only_creation_date) { EDTF.parse('2020') }
+    let(:work) { build(:work, created_edtf: year_only_creation_date) }
+
+    it 'renders the component without month or day selected' do
+      expect(rendered.css('#work_created_year').first['value']).to eq '2020'
+      expect(rendered.css('#work_created_month option[@selected="selected"]')).to match_array([])
```
Comment (mjgiarlo) [inline spec/components/works/dates_component_spec.rb:60]: Given it's an empty array in the expectation, I wonder how you'd feel about replacing `match_array([])` with `be_empty`

File: spec/components/works/dates_component_spec.rb [hunk 3]
```diff
@@ -50,4 +50,26 @@
...
+      expect(rendered.css('#work_created_day option[@selected="selected"]')).to match_array([])
```
Comment (mjgiarlo) [inline spec/components/works/dates_component_spec.rb:61]: See :point_up:

File: spec/components/works/dates_component_spec.rb [hunk 4]
```diff
@@ -50,4 +50,26 @@
...
+  end
+
+  context 'with a populated form containing year and month for single creation_date' do
+    let(:year_month_creation_date) { EDTF.parse('2020-05') }
+    let(:work) { build(:work, created_edtf: year_month_creation_date) }
+
+    it 'renders the component without day selected' do
+      expect(rendered.css('#work_created_year').first['value']).to eq '2020'
+      expect(rendered.css('#work_created_month option[@selected="selected"]').first['value']).to eq '5'
+      expect(rendered.css('#work_created_day option[@selected="selected"]')).to match_array([])
```
Comment (mjgiarlo) [inline spec/components/works/dates_component_spec.rb:72]: See :point_up:

File: spec/components/works/publication_date_component_spec.rb [hunk 5]
```diff
@@ -21,4 +21,26 @@
...
     end
   end
+
+  context 'with a populated form containing only year for publication_date' do
+    let(:year_only_pub_date) { EDTF.parse('2020') }
+    let(:work) { build(:work, published_edtf: year_only_pub_date) }
+
+    it 'renders the component without month or day selected' do
+      expect(rendered.css('#work_published_year').first['value']).to eq '2020'
+      expect(rendered.css('#work_published_month option[@selected="selected"]')).to match_array([])
```
Comment (mjgiarlo) [inline spec/components/works/publication_date_component_spec.rb:31]: See :point_up:

File: spec/components/works/publication_date_component_spec.rb [hunk 6]
```diff
@@ -21,4 +21,26 @@
...
+      expect(rendered.css('#work_published_day option[@selected="selected"]')).to match_array([])
```
Comment (mjgiarlo) [inline spec/components/works/publication_date_component_spec.rb:32]: See :point_up:

File: spec/components/works/publication_date_component_spec.rb [hunk 7]
```diff
@@ -21,4 +21,26 @@
...
+  end
+
+  context 'with a populated form containing year and month for publication_date' do
+    let(:year_month_pub_date) { EDTF.parse('2020-05') }
+    let(:work) { build(:work, published_edtf: year_month_pub_date) }
+
+    it 'renders the component without day selected' do
+      expect(rendered.css('#work_published_year').first['value']).to eq '2020'
+      expect(rendered.css('#work_published_month option[@selected="selected"]').first['value']).to eq '5'
+      expect(rendered.css('#work_published_day option[@selected="selected"]')).to match_array([])
```
Comment (mjgiarlo) [inline spec/components/works/publication_date_component_spec.rb:43]: See :point_up:

**Review by mjgiarlo — APPROVED** [review]
LGTM! I have a question for you about the specs, but it is not terribly important.

File: spec/components/works/dates_component_spec.rb (hunk 2 above)
Comment (jermnelson) [reply spec/components/works/dates_component_spec.rb:60]: Like it! Changed.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**